pkg/pillar/cmd/zedmanager/zedmanager.go

```diff
@@ -390,6 +399,26 @@ func checkRetry(ctxPtr *zedmanagerContext) {
 	}
 }
 
+// Handle the applications in the START_DELAY state ready to be started.
+func checkDelayedStartApps(ctx *zedmanagerContext) {
+	configs := ctx.subAppInstanceConfig.GetAll()
+	for _, c := range configs {
+		config := c.(types.AppInstanceConfig)
+		status := lookupAppInstanceStatus(ctx, config.Key())
+		// Is the application in the delayed state and ready to be started?
+		if status != nil && status.State == types.START_DELAYED && status.StartTime.Before(time.Now()) {
+			// Do not handle the application that is already being handled.
+			// It protects from the case when the application has already been started for handling but has not  changed
+			// its state yet.
```

Comment:
It sets the state to BOOTING at the very end of the function, inside the doActivateTail() call.
Also, it's possible that we even do not get to the doActivate() call, as from the checkDelayedStartApps() function we do not call it directly but call a more generic handler: doUpdate().
So, relying on a state is not a good option here, I think.

Reply:
If we delayed and then when the delay was over we call doUpdate() and that does not change the state from START_DELAY to some other state, then 1) something is clearly wrong and 2) we better try again by calling doUpdate() again otherwise it would incorrectly be stuck in START_DELAY forever. So I think this additional boolean should be removed and instead only rely on the START_DELAY state plus the check that more that the delay time has passed. 
Or am I missing something where we want to leave it in START_DELAY state? Only possibility I can see is that somewhere in doUpdate or doActivate we got an error in which case we'd want the boot retry timer to kick in later to try again, but in that case it doesn't make sense to leave the state at START_DELAY.

Reply:
In theory, instead of setting the `delayHanlded` flag, I can change the state to something other than `START_DELAYED`. The question then is - what is the right state that we should set. `BOOTING`?

Reply:
I expect the state to be INSTALLED but as I said above you should't set START_DELAY until after the networking checks.
So setting it back to INSTALLED should work.